ipatests/test_integration/test_nfs.py

```diff
@@ -0,0 +1,252 @@
+   krb5 NFS and automount locations.
+
+   Wishlist
+   * add automount direct and indirect maps
+   * add automount /home for the "seattle" location only
+   * validate it is not available in another location
+   * krb5 /home for IdM users in test_krb5_nfs_manual_configuration
+   * krb5 /home for IdM users in test_automount_location
+   * better validation of "ipa automountlocation-tofiles" output
+   * store nfs configuration in a single place
+"""
+
+from __future__ import absolute_import
+
+import time
+import re
+
+from ipatests.test_integration.base import IntegrationTest
+from ipatests.pytest_ipa.integration import tasks
+from ipaplatform.paths import paths
+
+import os
+
+# give some time for units to stabilize
+# otherwise we get transient errors
+WAIT_AFTER_INSTALL = 5
+WAIT_AFTER_UNINSTALL = WAIT_AFTER_INSTALL
+
+
+class TestInit(IntegrationTest):
+
+    @classmethod
+    def fix_resolv_conf(cls, client, server):
+
+        contents = client.get_file_contents(paths.RESOLV_CONF,
+                                            encoding='utf-8')
+        nameserver = 'nameserver %s\n' % server.ip
+        if not contents.startswith(nameserver):
+            contents = nameserver + contents.replace(nameserver, '')
+            client.put_file_contents(paths.RESOLV_CONF, contents)
+
+
+class TestNFS(TestInit):
+
+    num_replicas = 2
+    num_clients = 1
+    topology = 'star'
+
+    @classmethod
+    def install(cls, mh):
+
+        tasks.install_master(cls.master, setup_dns=True)
+        clients = (cls.clients[0], cls.replicas[0], cls.replicas[1])
+        for client in clients:
+            cls.fix_resolv_conf(client, cls.master)
+            tasks.install_client(cls.master, client)
+            client.run_command(["cat", "/etc/resolv.conf"])
+
+    def cleanup(self):
+
+        nfssrv = self.clients[0]
+        nfsclt = self.replicas[0]
+        automntclt = self.replicas[1]
+
+        nfsclt.run_command(["umount", "-a", "-t", "nfs4"])
+        nfsclt.run_command(["systemctl", "stop", "rpc-gssd"])
+
+        nfssrv.run_command(["systemctl", "stop", "nfs-server"])
+        nfssrv.run_command(["systemctl", "disable", "nfs-server"])
+        nfssrv.run_command([
+            "rm", "-f", "/etc/exports.d/krbnfs.exports",
+            "/etc/exports.d/stdnfs.exports"
+        ])
+
+        nfssrv.run_command(["rm", "-rf", "/exports"])
+
+        tasks.uninstall_client(nfsclt)
+        tasks.uninstall_client(nfssrv)
+        self.master.run_command([
+            "ipa", "host-mod", automntclt.hostname,
+            "--location", "''"
+        ])
+        # not strictly necessary, but this exercises automountlocation-del
+        self.master.run_command([
+            "ipa", "automountlocation-del", "seattle"
+        ])
+        nfsclt.run_command(["systemctl", "restart", "nfs-utils"])
+        nfssrv.run_command(["systemctl", "restart", "nfs-utils"])
+        # TODO put /etc/resolv.conf back
```

Comment:
Is it possible to implement also this TODO?

Reply:
Done. If you want to confirm, please search for resolv.conf in http://freeipa-org-pr-ci.s3-website.eu-central-1.amazonaws.com/jobs/e009d794-5fc8-11e9-b1f8-fa163e937cf0/report.html